presto-main/src/test/java/com/facebook/presto/operator/TestLookupJoinPageBuilder.java

```diff
@@ -0,0 +1,209 @@
+            }
+        }
+        assertTrue(lookupJoinPageBuilder.toString().contains("positionCount=" + output.getPositionCount()));
+
+        lookupJoinPageBuilder.reset();
+        assertTrue(lookupJoinPageBuilder.isEmpty());
+    }
+
+    private final class TestLookupSource
+            implements LookupSource
+    {
+        private final List<Type> types;
+        private final Page page;
+
+        public TestLookupSource(List<Type> types, Page page)
+        {
+            this.types = types;
+            this.page = page;
+        }
+
+        @Override
+        public int getChannelCount()
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public long getJoinPositionCount()
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public long joinPositionWithinPartition(long joinPosition)
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public long getInMemorySizeInBytes()
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public long getJoinPosition(int position, Page page, Page allChannelsPage, long rawHash)
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public long getJoinPosition(int position, Page hashChannelsPage, Page allChannelsPage)
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public long getNextJoinPosition(long currentJoinPosition, int probePosition, Page allProbeChannelsPage)
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public boolean isJoinPositionEligible(long currentJoinPosition, int probePosition, Page allProbeChannelsPage)
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public void appendTo(long position, PageBuilder pageBuilder, int outputChannelOffset)
+        {
+            for (int i = 0; i < types.size(); i++) {
+                types.get(i).appendTo(page.getBlock(i), (int) position, pageBuilder.getBlockBuilder(i));
+            }
+        }
+
+        @Override
+        public void close()
+        {
+        }
+    }
+
+    private final class TestJoinProbe
+            implements JoinProbe
+    {
+        private final Page page;
+        private int position = 0;
+
+        public TestJoinProbe(Page page)
+        {
+            this.page = page;
+        }
+
+        @Override
+        public int getOutputChannelCount()
+        {
+            return page.getChannelCount();
+        }
+
+        @Override
+        public int[] getOutputChannels()
+        {
+            int[] channels = new int[getOutputChannelCount()];
+            setAll(channels, i -> i);
+            return channels;
```

Comment:
Production code doesn't copy in `getOutputChannels()` so test code shouldn't do that either.